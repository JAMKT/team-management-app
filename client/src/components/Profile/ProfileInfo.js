import React, { useState, useCallback, useEffect } from 'react';
import { BrowserRouter, Route, Switch, Link, Redirect } from "react-router-dom";
import axios from 'axios';
import UserCardVertical from "../Common/UserCard/UserCardVertical";

const ProfileInfo = (props) => {
    const [userInfo, setUserInfo] = useState(null);
    var userData = {};

    if(props.user){
        userData = props.user;
    }
    
    useEffect(()=>{
        if(userInfo == null ){
            setUserInfo(props.user);
        }
    });
  
    return (
        <div id="profile-info" className="col card col-fixed-372">
            <div className="row justify-end">
                <button className="edit-btn">Edit</button>
            </div>
            <UserCardVertical user={userInfo}/>
            <div className="row">
                <h3>Who am I?</h3>
            </div>
                <div className="row">
                <p>{userData.description}</p>
            </div>
            <div className="row">
                <h3>Contact information</h3>
            </div>
            <div className="row">
                <div className="col">
                    <div className="row">
                        <span>{userData.email}</span>
                    </div>
                    <div className="row">
                        <span>{userData.phoneNumber}</span>
                    </div>
                </div>
            </div>
            <div className="row">
                <h3>Workload</h3>
            </div>
            <div className="row">
                
                Workload Widget...
                
            </div>
        </div>
    );
};

export default ProfileInfo;