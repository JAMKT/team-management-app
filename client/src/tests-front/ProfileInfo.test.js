import { shallow } from 'enzyme';
import React from 'react';
import ProfileInfo from '../components/Profile/ProfileInfo.js';

//Testing if the ProfileInfo component is rendered
it('expect to render ProfileInfo component', () => {
    expect(shallow(<ProfileInfo />).length).toEqual(1)
})

//Take a snapshot of a component to track changes
//The test will fail if the changes are made in ProfileInfo component
//If you intended those changes, press u if the test fails to update the snapshot
it('expect to render ProfileInfo component (make a snapshot)', () => {
    expect(shallow(<ProfileInfo />).debug()).toMatchSnapshot();
})