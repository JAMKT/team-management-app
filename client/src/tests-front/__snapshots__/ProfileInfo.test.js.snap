// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render ProfileInfo component (make a snapshot) 1`] = `
"<div id=\\"profile-info\\" className=\\"col card col-fixed-372\\">
  <div className=\\"row justify-end\\">
    <button className=\\"edit-btn\\">
      Edit
    </button>
  </div>
  <UserCardVertical user={{...}} />
  <div className=\\"row\\">
    <h3>
      Who am I?
    </h3>
  </div>
  <div className=\\"row\\">
    <p />
  </div>
  <div className=\\"row\\">
    <h3>
      Contact information
    </h3>
  </div>
  <div className=\\"row\\">
    <div className=\\"col\\">
      <div className=\\"row\\">
        <span />
      </div>
      <div className=\\"row\\">
        <span />
      </div>
    </div>
  </div>
  <div className=\\"row\\">
    <h3>
      Workload
    </h3>
  </div>
  <div className=\\"row\\">
    Workload Widget...
  </div>
</div>"
`;
